combat-trainer.lic

```diff
@@ -1999,6 +2036,62 @@ class GameState
     @attack_overrides[weapon_skill]
   end
 
+  def set_currently_whirlwinding(setting)
+    @currently_whirlwinding = setting
+  end
+
+  def currently_whirlwinding?
+    @currently_whirlwinding
+  end
+
+  def whirlwind_trainables
+    @whirlwind_trainables
+  end
+
+  def whirlwind_offhand_name
+    weapon_training[@current_whirlwind_offhand_skill]
+  end
+
+  def last_whirlwind_offhand_weapon_name
+    weapon_training[@last_whirlwind_offhand_skill]
+  end
+
+  def update_whirlwind_weapon_info(weapon_skill)
+    @last_whirlwind_offhand_skill = @current_whirlwind_offhand_skill
+    @current_whirlwind_offhand_skill = weapon_skill
+  end
+
+  def whirlwind_offhand_skill
+    @current_whirlwind_offhand_skill
+  end
+
+  def is_whirlwind_trainable?(weapon_skill)
+    return false if !whirlwind_trainables.any?
+    whirlwind_trainables.include? (weapon_skill)
+  end
+
+  def sheath_whirlwind_offhand
+    return unless currently_whirlwinding?
+    @equipment_manager.stow_weapon(whirlwind_offhand_name)
+  end
+
+  def wield_whirlwind_offhand
+    return unless currently_whirlwinding?
+    @equipment_manager.wield_weapon_offhand(whirlwind_offhand_name)
+  end
+
+  def determine_whirlwind_action
+    if @use_weak_attacks && Flags["ct-accuracy-ready"]
+      update_analyze_array(true) if @analyze_combo_array.empty?
+      return @analyze_combo_array.shift
+    elsif (npcs.length > 2) && !balance_low? 
+      return 'whirlwind' 
+    elsif balance_low? 
+      return 'gouge'
```

Comment:
Yeah, learned that from Cainec. All of these things that put me off from getting this setup, turned out to be things that weren't very difficult to handle or non issues.